### PR TITLE
docs: docusaurus based documentation

### PR DESCRIPTION
**Pull request by zklosko**
### Description

A current pain point with our documentation site is a lack of multilingual support. This PR switches us to the [Docusaurus](https://docusaurus.io/) static site framework, giving us first-class multilingual support, built-in search with autocompletion, better syntax highlighting, better docs organization, and CI/CD with Github Actions. Docs can be written in Markdown and MDX (using React); additionally, standalone pages can be written in JSX.

Two things to be aware of:

- The Github Actions workflow builds the site to a `gh-pages` branch of the repo, which only contains the output of `Docusaurus build`. The site is served from the `gh-pages` branch.
- Docusaurus uses [Infima](https://infima.dev/), a CSS framework being developed alongside Docusaurus. We would be using the Docusaurus 2 beta, and the developers recommend we keep theme adjustments to a minimum as [future changes around theming could be breaking changes](https://docusaurus.io/docs/advanced/swizzling#swizzling-1) (see caution box). Version 1 has been depreciated and the developers strongly recommend use of version 2 beta over version 1.

A brief style guide for future documentation can be found at `/docs/README.md`.

### Testing Notes

Enter the docs folder and run `npm ci` (install from dependencies listed in package-lock file), then `npm run start` to start a local server or `npm run build` to build. Builds land in the build directory.

A live build can be found on [my repo](https://zklosko.github.io/libretime/).

### **Links**

Contributes to #1307 and #1447.

### CODE REVIEW

File: docs/docs/getting-started/reverse-proxy.md
Comment (jooola) [inline docs/docs/getting-started/reverse-proxy.md:77]: Please keep the Heredoc EOF marker here

File: docs/docs/getting-started/set-server-time.md
Comment (jooola) [inline docs/docs/getting-started/set-server-time.md:3]: I don't think this has to be in first position. Install and configuration should be first, this could go in an advanced config section or such.

File: docs/docs/getting-started/ssl.md
Comment (jooola) [inline docs/docs/getting-started/ssl.md:45]: I also think we should not rewrite documentation for other software, certbot has an good documentation let's delegate that to them.

File: docs/docs/getting-started/ssl.md
Comment (jooola) [inline docs/docs/getting-started/ssl.md:89]: - I think the codeblocks support apache syntax highlighting
- Can we import code blocks from files ? I would like to move this to a dedicated file.

Comment (jooola) [inline docs/docs/getting-started/ssl.md:89]: - I think the codeblocks support apache syntax highlighting
- Can we import code blocks from files ? I would like to move this to a dedicated file.

File: docs/docs/getting-started/ssl.md
Comment (jooola) [inline docs/docs/getting-started/ssl.md:91]: Yup, the docs do need some freshness. We don't support sysV anymore.

File: docs/docs/intro.md
Comment (jooola) [inline docs/docs/intro.md:20]: I feel this might also introduce too much duplication.

File: docs/docs/local-dev.md
Comment (jooola) [inline docs/docs/local-dev.md:5]: Isn't vagrant is our first class dev tool currently ? I would put mutipass below the vagrant documentation.

File: docs/docs/local-dev.md
Comment (jooola) [inline docs/docs/local-dev.md:23]: Ideally we could replace this with a link to the install documentation.

File: docs/docs/local-dev.md
Comment (jooola) [inline docs/docs/local-dev.md:48]: Possible missing closing triple colons

File: docs/docs/server-config/backing-up-the-server.md
Comment (jooola) [inline docs/docs/server-config/backing-up-the-server.md:8]: Not to myself: I don't like this statement, watched folder is not the same as a backup/restore mechanism, we should correct this.

File: docs/docs/server-config/host-configuration.md
Comment (jooola) [inline docs/docs/server-config/host-configuration.md:18]: Please rebase

File: docs/docs/server-config/icecast.md
Comment (jooola) [inline docs/docs/server-config/icecast.md:34]: Missing syntax highlighting

File: docs/docs/upgrading.md
Comment (jooola) [inline docs/docs/upgrading.md:44]: Please rebase.

File: docs/docusaurus.config.js
Comment (jooola) [inline docs/docusaurus.config.js:14]: Why not throw ?

File: docs/docusaurus.config.js
Comment (jooola) [inline docs/docusaurus.config.js:11]: Should be updated, and CNAME should be restored as well.

File: docs/docusaurus.config.js
Comment (jooola) [inline docs/docusaurus.config.js:32]: I saw some English files in the i18n/fr/** dir. Those should probably be moved.

File: docs/package.json
Comment (jooola) [inline docs/package.json:21]: I think we should use classnames.

File: docs/src/pages/contribute.md
Comment (jooola) [inline docs/src/pages/contribute.md:12]: Please rebase.

File: docs/blog/authors.yml
Comment (jooola) [inline docs/blog/authors.yml:6]: Let's let Lucas add his own name when he wants.

**Review by jooola — CHANGES_REQUESTED** [review]
Globally I think we should make this PR as slim as possible, future improvement can be made in other PRs.

Made a small review so far.

Some extra things to consider:
- While we are moving all the files around, we should move all the documentation pictures next to the markdown file using it. This should make it easier to edit. (maybe this can go in another PR)
- I would like to keep the files git history, so after this is ready to be merged I would like to rewrite the commit history.
- Could we move the documentation in the /docs folder and put the website + blog in a /website folder ?
- For another PR, I think we should make a clear separation of the user documentation and the admin documentation.

File: docs/docusaurus.config.js
Comment (zklosko) [reply docs/docusaurus.config.js:32]: Those files have been removed.

File: docs/package.json
Comment (zklosko) [reply docs/package.json:21]: We do. Clsx is used by the classic Docusaurus theme.

File: .github/workflows/test-deploy-docs.yml
Comment (jooola) [inline .github/workflows/test-deploy-docs.yml:23]: I don't think we need a test deployment.

File: docs/docs/getting-started/install.md
Comment (jooola) [reply docs/docs/getting-started/install.md:10]: Please remove the note entirely.

File: docs/docs/getting-started/install.md
Comment (jooola) [inline docs/docs/getting-started/install.md:17]: Both previous and new cpu requirements are really vague, but 1Ghz feel more like a minimum requirement as "A modern, multi-core processor", people will always be able to scale up if speed is required.

File: docs/docs/getting-started/reverse-proxy.md
Comment (jooola) [inline docs/docs/getting-started/reverse-proxy.md:42]: We don't want to go back to the intro (kind of an entrypoint) when we are already deep in advanced configuration, I would head to the complete install documentation page.

File: docs/docs/getting-started/set-server-time.md
Comment (jooola) [reply docs/docs/getting-started/set-server-time.md:3]: Why did you mark this as resolved?

File: docs/docs/getting-started/ssl.md
Comment (jooola) [inline docs/docs/getting-started/ssl.md:51]: Please use the apache syntax highlighting using the `apacheconf` keyword and add it to the additional languages in the prism config https://docusaurus.io/docs/markdown-features/code-blocks#supported-languages.

File: docs/docs/server-config/host-configuration.md
Comment (jooola) [inline docs/docs/server-config/host-configuration.md:13]: This one could use the `ini` code block syntax highlighting

File: docs/docs/server-config/host-configuration.md
Comment (jooola) [reply docs/docs/server-config/host-configuration.md:18]: Not yet solved.

File: docs/docs/server-config/icecast.md
Comment (jooola) [inline docs/docs/server-config/icecast.md:50]: This is xml, not html.

File: docs/docs/server-config/icecast.md
Comment (jooola) [inline docs/docs/server-config/icecast.md:100]: This is xml, not html.

**Review by jooola — CHANGES_REQUESTED** [review]

**Comment by jooola** [comment]
I suggest we don't include the releases notes yet, it can be handled in another PR. I already started to prepare some files in #1568 and waiting for this PR to be merged before continuing.

This PR is gonna refreshing ! Thanks for the work.

**Comment by paddatrapper** [comment]
Currently our docs are built and deployed from https://github.com/LibreTime/LibreTime.github.io using TravisCI and I've just noticed that it hasn't worked in a year. Is it possible to build and create a PR that is auto-merged or a git push to that repo? Otherwise we will need to change the DNS configuration, which only @hairmare has access to.

**Comment by jooola** [comment]
> Currently our docs are built and deployed from https://github.com/LibreTime/LibreTime.github.io using TravisCI and I've just noticed that it hasn't worked in a year. Is it possible to build and create a PR that is auto-merged or a git push to that repo? Otherwise we will need to change the DNS configuration, which only @hairmare has access to.

I was confused at first, but I think Github automagically takes the docs folder and build it using jekyll. Jekyll being the default github page engine.

**Comment by paddatrapper** [comment]
Ah, yes you're right - the DNS is already configured to point to this repo and not the old docs repo. That should be archived

**Comment by jooola** [comment]
> Ah, yes you're right - the DNS is already configured to point to this repo and not the old docs repo. That should be archived

I don't know if we can move the one issue on that repo once it is archived.

File: docs/README.md
Comment (zklosko) [reply docs/README.md:7]: I ran into issues with the Github Actions worker and yarn, but I can try again.

File: docs/docs/getting-started/ssl.md
Comment (zklosko) [reply docs/docs/getting-started/ssl.md]: Resolving the duplicate comment

File: docs/package.json
Comment (jooola) [reply docs/package.json:21]: I meant using https://github.com/JedWatson/classnames instead of clsx, the latter seem unmaintained.

File: .github/workflows/test-deploy-docs.yml
Comment (zklosko) [reply .github/workflows/test-deploy-docs.yml:23]: Fair

File: docs/blog/2021-10-15-3.0-alpha-10-released.md
Comment (zklosko) [reply docs/blog/2021-10-15-3.0-alpha-10-released.md]: Okay, I'll remove the blog for now. We can add it back in a later PR.

By the way, Docusaurus can manage multiple blogs and docs instances in addition to a regular React-powered site, so there's no need to have separate folders or generators.

**Comment by zklosko** [comment]
> I suggest we don't include the releases notes yet, it can be handled in another PR. I already started to prepare some files in https://github.com/LibreTime/libretime/pull/1568 and waiting for this PR to be merged before continuing.

I think we should at least get the latest release notes in there. What you're putting together looks more like a blog section and may not fit snugly with this PR.

File: docs/package.json
Comment (zklosko) [reply docs/package.json:21]: Since Docusaurus is still under active development, I'd say leave that detail to the devs since it ships with the base theme. There aren't really any third party themes at the moment.

File: .github/workflows/deploy-docs.yml
Comment (jooola) [inline .github/workflows/deploy-docs.yml:7]: Let's use this workflow for all that is website/docs related.

Could you rename the file to `website.yml` or `docs.yml` and update the worklfow name accordingly. This workflow with do more than simply deploy the website at some point.

You probably thought about this already, just marking it so we don't forget: branches needs to be updated

File: .github/workflows/deploy-docs.yml
Comment (jooola) [inline .github/workflows/deploy-docs.yml:16]: Let's use node current lts `16.x`

File: docs/README.md
Comment (jooola) [reply docs/README.md:7]: https://github.com/peaceiris/actions-gh-pages#%EF%B8%8F-docusaurus

File: docs/docs/server-config/icecast.md
Comment (jooola) [inline docs/docs/server-config/icecast.md:29]: This is xml, not html.

File: docs/docs/server-config/icecast.md
Comment (jooola) [inline docs/docs/server-config/icecast.md:111]: This is xml, not html.

**Review by jooola — CHANGES_REQUESTED** [review]

**Comment by jooola** [comment]
Legacy linting failure is fixed in #1575 

@zklosko I pushed a few changes. I think we are almost there. Only a few things topics to resolve and we will be able to merge this. Let me know when you think your work is done in this PR.

**Comment by paddatrapper** [comment]
As part of the doc linting, perhaps we should include something like https://github.com/gaurav-nelson/github-action-markdown-link-check?

**Comment by jooola** [comment]
> As part of the doc linting, perhaps we should include something like https://github.com/gaurav-nelson/github-action-markdown-link-check?

Docusaurus seem to already handle broken links https://docusaurus.io/docs/api/docusaurus-config#onbrokenlinks and https://docusaurus.io/docs/api/docusaurus-config#onbrokenmarkdownlinks

File: docs/guides/scheduling-shows.md
Comment (paddatrapper) [inline docs/guides/scheduling-shows.md:38]:
```suggestion
The track scheduler behaves similarly to iTunes or Windows Media Player: media browser on the left, playlist on the right.
```

File: docs/guides/scheduling-shows.md
Comment (paddatrapper) [inline docs/guides/scheduling-shows.md:39]:
```suggestion
Find the tracks that you would like to schedule by using the search box or sorting columns and then dragging them
```

File: docs/guides/users.md
Comment (paddatrapper) [inline docs/guides/users.md:7]:
```suggestion
It is strongly recommended not to use the default `admin` account in production, especially if your LibreTime server is accessible from the internet.
```

File: docs/server-config/icecast.md
Comment (paddatrapper) [inline docs/server-config/icecast.md:7]:
```suggestion
LibreTime supports direct connection to two popular streaming media servers, the open source Icecast (<http://www.icecast.org>) and the proprietary SHOUTcast (<http://www.shoutcast.com>). Apart from the software license, the main difference between these two servers is that Icecast supports simultaneous MP3, AAC, Ogg Vorbis or Ogg Opus streaming from LibreTime, whereas SHOUTcast only supports MP3 and AAC streams. The royalty-free Ogg Vorbis format has the advantage of better sound quality than MP3 at lower bitrates, which has a direct impact on the amount of bandwidth that your station will require to serve the same number of listeners. Ogg Opus also benefits from good sound quality at low bitrates, with the added advantage of lower latency than other streaming formats. Opus is now an [IETF standard](http://tools.ietf.org/html/rfc6716) and requires Icecast 2.4 or later to be installed on the streaming server.
```

File: website/docusaurus.config.js
Comment (jooola) [reply website/docusaurus.config.js:12]: This should vanish before merging anyway, I'll make sure of it.

**Comment by jooola** [comment]
According to https://github.com/facebook/docusaurus/pull/3059#issue-656778130 external links are not checked by docusaurus. I'll setup a tool that scan for this.

File: website/docusaurus.config.js
Comment (zklosko) [reply website/docusaurus.config.js:12]: Correct. This is only set this way so it publishes correctly in my fork.

**Comment by zklosko** [comment]
@jooola I'll let you get the external link checker. I would want to design out the home page a little more and then I'll be done. Thanks for helping me fix up the place (@paddatrapper too).

**Comment by zklosko** [comment]
@jooola Only other thing I can think of is to request that you let me fold in #1568 as part of this PR to make it easier

**Comment by jooola** [comment]
@zklosko As I proposed initially, we should keep this PR as slim as possible and only do small steps at the time. This PR is already really large and I would not want to squeeze any extra change in there.

The commit history is something we should really try to preserve as it is a huge value for people in the future working in this project. Having too much change in a single PR, make it difficult to keep it clean.

I'd be happy to review any future PR, just branch again and open a new PR.

I propose to fix the few little paths and names that needs some update, squash the history, and merge this.

**Comment by jooola** [comment]
And let's keep the old pictures for now, we can still update or remove them later. Those are still valuables when improving the documentation.

**Comment by jooola** [comment]
@paddatrapper Are good to merge ? Any last comments ?

File: docs/_docs/interface-localization.md
Comment (jooola) [inline docs/_docs/interface-localization.md:5]: Seems like this file vanished from the new documentation.

File: docs/_docs/troubleshooting.md
Comment (jooola) [inline docs/_docs/troubleshooting.md:5]: Seems like this file vanished from the new documentation.

**Comment by jooola** [comment]
I rewrote the commit history on my local branch  with a few tweaks:
- resurrected the files that were missing
- added typescript capabilities the website
- a few tweaks here and there
- moved unused pictures in _backup folder.

See https://github.com/jooola/libretime/commits/lt-docusarus

@zklosko I would like to push and merge, but I don't want to force push on this branch without you approval.

File: docs/_docs/interface-localization.md
Comment (zklosko) [reply docs/_docs/interface-localization.md:5]: Yes, I've deleted the file

File: docs/_docs/troubleshooting.md
Comment (zklosko) [reply docs/_docs/troubleshooting.md:5]: Yes, I've deleted the file

**Comment by zklosko** [comment]
@jooola it looks good to me! I can do the homepage tweaks in a different PR, so I'd say it's ready to merge.

File: docs/_docs/interface-localization.md
Comment (jooola) [reply docs/_docs/interface-localization.md]: In the future, please highlight such removal and describe why you think it should happen.

**Review by jooola — APPROVED** [review]
Some changes could have used a dedicated PR, but I don't be too oppressive.

We should try to split code changes from content changes so it easier for us to review in the future. I would not want to loose/update content without minimal discussion.

I might dig into the old mkdocs documentation to see if anything can be resurrected from there.

Gonna merge this, so we don't block other PRs.

**Comment by jooola** [comment]
@paddatrapper We might need to tweak the repo settings to properly handle the gh-page deployement

**Comment by paddatrapper** [comment]
Updated, but still 404 response. Pointing pages to gh-pages branch, root directory

**Pull request merged**
